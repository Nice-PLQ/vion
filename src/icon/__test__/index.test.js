import { mount } from '@vue/test-utils';
import Icon from '../icon.vue';

describe('Icon', () => {
  test('图标显示', async () => {
    const wrapper = mount(Icon, {
      props: {
        name: 'AddCircle',
        color: '#000',
        size: '24px',
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('v-icon');
    expect(wrapper.element.style.color).toBe('rgb(0, 0, 0)');
    expect(wrapper.element.style.fontSize).toBe('24px');

    await wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
  });
});
