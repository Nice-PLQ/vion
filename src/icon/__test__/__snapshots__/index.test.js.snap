// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 图标显示 1`] = `
<i
  class="v-icon"
  style="display: inline-block; font-size: 24px; color: rgb(0, 0, 0); width: 1em; height: 1em; line-height: 1em;"
>
  <svg
    fill="currentColor"
    height="1em"
    viewBox="0 0 48 48"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M24 8C32.8366 8 40 15.1634 40 24C40 32.8366 32.8366 40 24 40C15.1634 40 8 32.8366 8 24C8 15.1634 15.1634 8 24 8ZM24 4C35.0457 4 44 12.9543 44 24C44 35.0457 35.0457 44 24 44C12.9543 44 4 35.0457 4 24C4 12.9543 12.9543 4 24 4ZM22 22V14H26V22H34V26H26V34H22V26H14V22H22Z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</i>
`;
